src/Symfony/Component/RateLimiter/TokenBucket.php

```diff
@@ -70,15 +70,39 @@ public function getExpirationTime(): int
         return $this->rate->calculateTimeForTokens($this->burstSize);
     }
 
-    public function serialize(): string
+    /**
+     * @internal
+     */
+    public function __serialize(): array
```

Comment:
not needed since __wakeup/__sleep do the job, isn't it? (same for __unser)

Reply:
Sorry, I misunderstood then (thought you wanted wakeup/sleep as a forward-compatibility). Removed the serialize magic methods.